### ISSUE TITLE
some error has happened when build OpenBLAS

### ISSUE THREAD

**Issue by shpeng123** [xianyi/OpenBLAS#4555]
hey,when I try to install OpenBLAS,but some problem has happened.The details as bellow:
when execute the command 'make install',the error message as follows:
getarch_2nd.c: In function ‘main’:
getarch_2nd.c:22:35: error: ‘SGEMM_DEFAULT_UNROLL_M’ undeclared (first use in this function); did you mean ‘SBGEMM_DEFAULT_UNROLL_M’?
     printf("SGEMM_UNROLL_M=%d\n", SGEMM_DEFAULT_UNROLL_M);
                                   ^~~~~~~~~~~~~~~~~~~~~~
                                   SBGEMM_DEFAULT_UNROLL_M
getarch_2nd.c:22:35: note: each undeclared identifier is reported only once for each function it appears in
getarch_2nd.c:23:35: error: ‘SGEMM_DEFAULT_UNROLL_N’ undeclared (first use in this function); did you mean ‘SBGEMM_DEFAULT_UNROLL_N’?
     printf("SGEMM_UNROLL_N=%d\n", SGEMM_DEFAULT_UNROLL_N);
                                   ^~~~~~~~~~~~~~~~~~~~~~
                                   SBGEMM_DEFAULT_UNROLL_N
getarch_2nd.c:24:35: error: ‘DGEMM_DEFAULT_UNROLL_M’ undeclared (first use in this function); did you mean ‘XGEMM_DEFAULT_UNROLL_M’?
     printf("DGEMM_UNROLL_M=%d\n", DGEMM_DEFAULT_UNROLL_M);
                                   ^~~~~~~~~~~~~~~~~~~~~~
                                   XGEMM_DEFAULT_UNROLL_M
getarch_2nd.c:25:35: error: ‘DGEMM_DEFAULT_UNROLL_N’ undeclared (first use in this function); did you mean ‘XGEMM_DEFAULT_UNROLL_N’?
     printf("DGEMM_UNROLL_N=%d\n", DGEMM_DEFAULT_UNROLL_N);
                                   ^~~~~~~~~~~~~~~~~~~~~~
                                   XGEMM_DEFAULT_UNROLL_N
getarch_2nd.c:29:35: error: ‘CGEMM_DEFAULT_UNROLL_M’ undeclared (first use in this function); did you mean ‘XGEMM_DEFAULT_UNROLL_M’?
     printf("CGEMM_UNROLL_M=%d\n", CGEMM_DEFAULT_UNROLL_M);
                                   ^~~~~~~~~~~~~~~~~~~~~~
                                   XGEMM_DEFAULT_UNROLL_M
getarch_2nd.c:30:35: error: ‘CGEMM_DEFAULT_UNROLL_N’ undeclared (first use in this function); did you mean ‘XGEMM_DEFAULT_UNROLL_N’?
     printf("CGEMM_UNROLL_N=%d\n", CGEMM_DEFAULT_UNROLL_N);
                                   ^~~~~~~~~~~~~~~~~~~~~~
                                   XGEMM_DEFAULT_UNROLL_N
getarch_2nd.c:31:35: error: ‘ZGEMM_DEFAULT_UNROLL_M’ undeclared (first use in this function); did you mean ‘XGEMM_DEFAULT_UNROLL_M’?
     printf("ZGEMM_UNROLL_M=%d\n", ZGEMM_DEFAULT_UNROLL_M);
                                   ^~~~~~~~~~~~~~~~~~~~~~
                                   XGEMM_DEFAULT_UNROLL_M
getarch_2nd.c:32:35: error: ‘ZGEMM_DEFAULT_UNROLL_N’ undeclared (first use in this function); did you mean ‘XGEMM_DEFAULT_UNROLL_N’?
     printf("ZGEMM_UNROLL_N=%d\n", ZGEMM_DEFAULT_UNROLL_N);
                                   ^~~~~~~~~~~~~~~~~~~~~~
                                   XGEMM_DEFAULT_UNROLL_N
getarch_2nd.c:79:50: error: ‘SGEMM_DEFAULT_Q’ undeclared (first use in this function); did you mean ‘SBGEMM_DEFAULT_Q’?
     printf("#define SLOCAL_BUFFER_SIZE\t%ld\n", (SGEMM_DEFAULT_Q * SGEMM_DEFAULT_UNROLL_N * 4 * 1 *  sizeof(float)));
                                                  ^~~~~~~~~~~~~~~
                                                  SBGEMM_DEFAULT_Q
getarch_2nd.c:80:50: error: ‘DGEMM_DEFAULT_Q’ undeclared (first use in this function); did you mean ‘SBGEMM_DEFAULT_Q’?
     printf("#define DLOCAL_BUFFER_SIZE\t%ld\n", (DGEMM_DEFAULT_Q * DGEMM_DEFAULT_UNROLL_N * 2 * 1 *  sizeof(double)));
                                                  ^~~~~~~~~~~~~~~
                                                  SBGEMM_DEFAULT_Q
getarch_2nd.c:81:50: error: ‘CGEMM_DEFAULT_Q’ undeclared (first use in this function); did you mean ‘SBGEMM_DEFAULT_Q’?
     printf("#define CLOCAL_BUFFER_SIZE\t%ld\n", (CGEMM_DEFAULT_Q * CGEMM_DEFAULT_UNROLL_N * 4 * 2 *  sizeof(float)));
                                                  ^~~~~~~~~~~~~~~
                                                  SBGEMM_DEFAULT_Q
getarch_2nd.c:82:50: error: ‘ZGEMM_DEFAULT_Q’ undeclared (first use in this function); did you mean ‘SBGEMM_DEFAULT_Q’?
     printf("#define ZLOCAL_BUFFER_SIZE\t%ld\n", (ZGEMM_DEFAULT_Q * ZGEMM_DEFAULT_UNROLL_N * 2 * 2 *  sizeof(double)));
                                                  ^~~~~~~~~~~~~~~
                                                  SBGEMM_DEFAULT_Q
make: *** [Makefile.prebuild:106: getarch_2nd] Error 1

The detailed configuration are as follows：
![configuration](https://github.com/OpenMathLib/OpenBLAS/assets/112854890/e2da0239-bc97-4f24-89d7-65b8c2c04694)

**Comment by martin-frbg** [comment]
The build system does not recognize the "fake" cpu id generated by qemu - you probably used the TARGET= option when you ran `make` before the `make install` ? Please repeat it and any other build options on the `make install` line.

**Comment by shpeng123** [comment]
I tried the command "make TARGET=GENERIC" and " make clean && make TARGET=NEHALEM"，but not solve the problem。Is that means this cpu is not fit the OpenBLAS?

**Comment by martin-frbg** [comment]
"make clean && make TARGET=NEHALEM" should work, followed by "make TARGET=NEHALEM install"

**Comment by martin-frbg** [comment]
Did you get it working ? I can only assume that you missed the `make clean` after the first attempt, as it should build fine when you supply a target cpu name instead of relying on autodetection

**Comment by HernandoR** [comment]
i met simular error when trying to config for colmap by vcpkg, i guess it's something to do with :
```
CMake Error at cmake/prebuild.cmake:1429 (MESSAGE):
  Compiling getarch_2nd failed Change Dir:
  '/home/lz/Codes/vcpkg/buildtrees/openblas/x64-linux-dbg/getarch2_build/CMakeFiles/CMakeTmp'
  ```

I attached the log file as follows:
```
[1/2] "/home/lz/cmake-3.30.3-linux-x86_64/bin/cmake" -E chdir ".." "/home/lz/cmake-3.30.3-linux-x86_64/bin/cmake" "/home/lz/Codes/vcpkg/buildtrees/openblas/src/v0.3.27-2f0a3701cf.clean" "-G" "Ninja" "-DCMAKE_BUILD_TYPE=Release" "-DCMAKE_INSTALL_PREFIX=/home/lz/Codes/vcpkg/packages/openblas_x64-linux" "-DFETCHCONTENT_FULLY_DISCONNECTED=ON" "-DUSE_THREAD=OFF" "-DUSE_SIMPLE_THREADED_LEVEL3=OFF" "-DDYNAMIC_ARCH=OFF" "-DBUILD_WITHOUT_LAPACK=ON" "-DNOFORTRAN=ON" "-DBU=_" "-DCMAKE_MAKE_PROGRAM=/home/lz/Codes/vcpkg/downloads/tools/ninja/1.10.2-linux/ninja" "-DCMAKE_SYSTEM_NAME=Linux" "-DBUILD_SHARED_LIBS=OFF" "-DVCPKG_CHAINLOAD_TOOLCHAIN_FILE=/home/lz/Codes/vcpkg/scripts/toolchains/linux.cmake" "-DVCPKG_TARGET_TRIPLET=x64-linux" "-DVCPKG_SET_CHARSET_FLAG=ON" "-DVCPKG_PLATFORM_TOOLSET=external" "-DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_SYSTEM_PACKAGE_REGISTRY=ON" "-DCMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP=TRUE" "-DCMAKE_VERBOSE_MAKEFILE=ON" "-DVCPKG_APPLOCAL_DEPS=OFF" "-DCMAKE_TOOLCHAIN_FILE=/home/lz/Codes/vcpkg/scripts/buildsystems/vcpkg.cmake" "-DCMAKE_ERROR_ON_ABSOLUTE_INSTALL_DESTINATION=ON" "-DVCPKG_CXX_FLAGS= -DNEEDBUNDERSCORE" "-DVCPKG_CXX_FLAGS_RELEASE=" "-DVCPKG_CXX_FLAGS_DEBUG=" "-DVCPKG_C_FLAGS= -DNEEDBUNDERSCORE" "-DVCPKG_C_FLAGS_RELEASE=" "-DVCPKG_C_FLAGS_DEBUG=" "-DVCPKG_CRT_LINKAGE=dynamic" "-DVCPKG_LINKER_FLAGS=" "-DVCPKG_LINKER_FLAGS_RELEASE=" "-DVCPKG_LINKER_FLAGS_DEBUG=" "-DVCPKG_TARGET_ARCHITECTURE=x64" "-DCMAKE_INSTALL_LIBDIR:STRING=lib" "-DCMAKE_INSTALL_BINDIR:STRING=bin" "-D_VCPKG_ROOT_DIR=/home/lz/Codes/vcpkg" "-D_VCPKG_INSTALLED_DIR=/home/lz/Codes/colmap/build/vcpkg_installed" "-DVCPKG_MANIFEST_INSTALL=OFF"
FAILED: ../CMakeCache.txt 
"/home/lz/cmake-3.30.3-linux-x86_64/bin/cmake" -E chdir ".." "/home/lz/cmake-3.30.3-linux-x86_64/bin/cmake" "/home/lz/Codes/vcpkg/buildtrees/openblas/src/v0.3.27-2f0a3701cf.clean" "-G" "Ninja" "-DCMAKE_BUILD_TYPE=Release" "-DCMAKE_INSTALL_PREFIX=/home/lz/Codes/vcpkg/packages/openblas_x64-linux" "-DFETCHCONTENT_FULLY_DISCONNECTED=ON" "-DUSE_THREAD=OFF" "-DUSE_SIMPLE_THREADED_LEVEL3=OFF" "-DDYNAMIC_ARCH=OFF" "-DBUILD_WITHOUT_LAPACK=ON" "-DNOFORTRAN=ON" "-DBU=_" "-DCMAKE_MAKE_PROGRAM=/home/lz/Codes/vcpkg/downloads/tools/ninja/1.10.2-linux/ninja" "-DCMAKE_SYSTEM_NAME=Linux" "-DBUILD_SHARED_LIBS=OFF" "-DVCPKG_CHAINLOAD_TOOLCHAIN_FILE=/home/lz/Codes/vcpkg/scripts/toolchains/linux.cmake" "-DVCPKG_TARGET_TRIPLET=x64-linux" "-DVCPKG_SET_CHARSET_FLAG=ON" "-DVCPKG_PLATFORM_TOOLSET=external" "-DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_SYSTEM_PACKAGE_REGISTRY=ON" "-DCMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP=TRUE" "-DCMAKE_VERBOSE_MAKEFILE=ON" "-DVCPKG_APPLOCAL_DEPS=OFF" "-DCMAKE_TOOLCHAIN_FILE=/home/lz/Codes/vcpkg/scripts/buildsystems/vcpkg.cmake" "-DCMAKE_ERROR_ON_ABSOLUTE_INSTALL_DESTINATION=ON" "-DVCPKG_CXX_FLAGS= -DNEEDBUNDERSCORE" "-DVCPKG_CXX_FLAGS_RELEASE=" "-DVCPKG_CXX_FLAGS_DEBUG=" "-DVCPKG_C_FLAGS= -DNEEDBUNDERSCORE" "-DVCPKG_C_FLAGS_RELEASE=" "-DVCPKG_C_FLAGS_DEBUG=" "-DVCPKG_CRT_LINKAGE=dynamic" "-DVCPKG_LINKER_FLAGS=" "-DVCPKG_LINKER_FLAGS_RELEASE=" "-DVCPKG_LINKER_FLAGS_DEBUG=" "-DVCPKG_TARGET_ARCHITECTURE=x64" "-DCMAKE_INSTALL_LIBDIR:STRING=lib" "-DCMAKE_INSTALL_BINDIR:STRING=bin" "-D_VCPKG_ROOT_DIR=/home/lz/Codes/vcpkg" "-D_VCPKG_INSTALLED_DIR=/home/lz/Codes/colmap/build/vcpkg_installed" "-DVCPKG_MANIFEST_INSTALL=OFF"
CMake Deprecation Warning at CMakeLists.txt:5 (cmake_minimum_required):
  Compatibility with CMake < 3.5 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.


-- The C compiler identification is Clang 18.1.8
-- The ASM compiler identification is Clang with GNU-like command-line
-- Found assembler: /usr/bin/clang
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/clang - skipped
-- Detecting C compile features
-- Detecting C compile features - done
CMake Warning at CMakeLists.txt:101 (message):
  CMake support is experimental.  It does not yet support all build options
  and may not produce the same Makefiles that OpenBLAS ships with.


CMake Warning (dev) at cmake/system_check.cmake:13 (if):
  Policy CMP0054 is not set: Only interpret if() arguments as variables or
  keywords when unquoted.  Run "cmake --help-policy CMP0054" for policy
  details.  Use the cmake_policy command to set the policy and suppress this
  warning.

  Quoted variables like "LINUX" will no longer be dereferenced when the
  policy is set to NEW.  Since the policy is not set the OLD behavior will be
  used.
Call Stack (most recent call first):
  cmake/system.cmake:8 (include)
  CMakeLists.txt:104 (include)
This warning is for project developers.  Use -Wno-dev to suppress it.

-- Looking for stdatomic.h
-- Looking for stdatomic.h - found
-- GEMM multithread threshold set to 4.
-- Running getarch
-- GETARCH results:
CORE=HASWELL
LIBCORE=haswell
NUM_CORES=22
HAVE_MMX=1
HAVE_SSE=1
HAVE_SSE2=1
HAVE_SSE3=1
HAVE_SSSE3=1
HAVE_SSE4_1=1
HAVE_SSE4_2=1
HAVE_AVX=1
HAVE_AVX2=1
HAVE_FMA3=1
MAKEFLAGS += -j 22

CMake Error at cmake/prebuild.cmake:1429 (MESSAGE):
  Compiling getarch_2nd failed Change Dir:
  '/home/lz/Codes/vcpkg/buildtrees/openblas/x64-linux-rel/getarch2_build/CMakeFiles/CMakeTmp'




  Run Build Command(s):
  /home/lz/Codes/vcpkg/downloads/tools/ninja/1.10.2-linux/ninja -v cmTC_eb93d

  [1/2] /usr/bin/clang -fPIC -DNEEDBUNDERSCORE -march=native
  -DGEMM_MULTITHREAD_THRESHOLD=4 -DNO_PARALLEL_MAKE=0
  -I"/home/lz/Codes/vcpkg/buildtrees/openblas/x64-linux-rel/getarch2_build"
  -I"/home/lz/Codes/vcpkg/buildtrees/openblas/src/v0.3.27-2f0a3701cf.clean"
  -I"/home/lz/Codes/vcpkg/buildtrees/openblas/x64-linux-rel" -MD -MT
  CMakeFiles/cmTC_eb93d.dir/getarch_2nd.c.o -MF
  CMakeFiles/cmTC_eb93d.dir/getarch_2nd.c.o.d -o
  CMakeFiles/cmTC_eb93d.dir/getarch_2nd.c.o -c
  /home/lz/Codes/vcpkg/buildtrees/openblas/src/v0.3.27-2f0a3701cf.clean/getarch_2nd.c


  FAILED: CMakeFiles/cmTC_eb93d.dir/getarch_2nd.c.o

  /usr/bin/clang -fPIC -DNEEDBUNDERSCORE -march=native
  -DGEMM_MULTITHREAD_THRESHOLD=4 -DNO_PARALLEL_MAKE=0
  -I"/home/lz/Codes/vcpkg/buildtrees/openblas/x64-linux-rel/getarch2_build"
  -I"/home/lz/Codes/vcpkg/buildtrees/openblas/src/v0.3.27-2f0a3701cf.clean"
  -I"/home/lz/Codes/vcpkg/buildtrees/openblas/x64-linux-rel" -MD -MT
  CMakeFiles/cmTC_eb93d.dir/getarch_2nd.c.o -MF
  CMakeFiles/cmTC_eb93d.dir/getarch_2nd.c.o.d -o
  CMakeFiles/cmTC_eb93d.dir/getarch_2nd.c.o -c
  /home/lz/Codes/vcpkg/buildtrees/openblas/src/v0.3.27-2f0a3701cf.clean/getarch_2nd.c



  /home/lz/Codes/vcpkg/buildtrees/openblas/src/v0.3.27-2f0a3701cf.clean/getarch_2nd.c:22:35:
  error: use of undeclared identifier 'SGEMM_DEFAULT_UNROLL_M'

     22 |     printf("SGEMM_UNROLL_M=%d\n", SGEMM_DEFAULT_UNROLL_M);
        |                                   ^

  
  /home/lz/Codes/vcpkg/buildtrees/openblas/src/v0.3.27-2f0a3701cf.clean/getarch_2nd.c:23:35:
  error: use of undeclared identifier 'SGEMM_DEFAULT_UNROLL_N'

     23 |     printf("SGEMM_UNROLL_N=%d\n", SGEMM_DEFAULT_UNROLL_N);
        |                                   ^

  
  /home/lz/Codes/vcpkg/buildtrees/openblas/src/v0.3.27-2f0a3701cf.clean/getarch_2nd.c:24:35:
  error: use of undeclared identifier 'DGEMM_DEFAULT_UNROLL_M'

     24 |     printf("DGEMM_UNROLL_M=%d\n", DGEMM_DEFAULT_UNROLL_M);
        |                                   ^

  
  /home/lz/Codes/vcpkg/buildtrees/openblas/src/v0.3.27-2f0a3701cf.clean/getarch_2nd.c:25:35:
  error: use of undeclared identifier 'DGEMM_DEFAULT_UNROLL_N'

     25 |     printf("DGEMM_UNROLL_N=%d\n", DGEMM_DEFAULT_UNROLL_N);
        |                                   ^

  
  /home/lz/Codes/vcpkg/buildtrees/openblas/src/v0.3.27-2f0a3701cf.clean/getarch_2nd.c:29:35:
  error: use of undeclared identifier 'CGEMM_DEFAULT_UNROLL_M'

     29 |     printf("CGEMM_UNROLL_M=%d\n", CGEMM_DEFAULT_UNROLL_M);
        |                                   ^

  
  /home/lz/Codes/vcpkg/buildtrees/openblas/src/v0.3.27-2f0a3701cf.clean/getarch_2nd.c:30:35:
  error: use of undeclared identifier 'CGEMM_DEFAULT_UNROLL_N'

     30 |     printf("CGEMM_UNROLL_N=%d\n", CGEMM_DEFAULT_UNROLL_N);
        |                                   ^

  
  /home/lz/Codes/vcpkg/buildtrees/openblas/src/v0.3.27-2f0a3701cf.clean/getarch_2nd.c:31:35:
  error: use of undeclared identifier 'ZGEMM_DEFAULT_UNROLL_M'

     31 |     printf("ZGEMM_UNROLL_M=%d\n", ZGEMM_DEFAULT_UNROLL_M);
        |                                   ^

  
  /home/lz/Codes/vcpkg/buildtrees/openblas/src/v0.3.27-2f0a3701cf.clean/getarch_2nd.c:32:35:
  error: use of undeclared identifier 'ZGEMM_DEFAULT_UNROLL_N'

     32 |     printf("ZGEMM_UNROLL_N=%d\n", ZGEMM_DEFAULT_UNROLL_N);
        |                                   ^

  
  /home/lz/Codes/vcpkg/buildtrees/openblas/src/v0.3.27-2f0a3701cf.clean/getarch_2nd.c:39:37:
  error: use of undeclared identifier 'SGEMM_DEFAULT_UNROLL_M'

     39 |     printf("CGEMM3M_UNROLL_M=%d\n", SGEMM_DEFAULT_UNROLL_M);
        |                                     ^

  
  /home/lz/Codes/vcpkg/buildtrees/openblas/src/v0.3.27-2f0a3701cf.clean/getarch_2nd.c:45:37:
  error: use of undeclared identifier 'SGEMM_DEFAULT_UNROLL_N'

     45 |     printf("CGEMM3M_UNROLL_N=%d\n", SGEMM_DEFAULT_UNROLL_N);
        |                                     ^

  
  /home/lz/Codes/vcpkg/buildtrees/openblas/src/v0.3.27-2f0a3701cf.clean/getarch_2nd.c:51:37:
  error: use of undeclared identifier 'DGEMM_DEFAULT_UNROLL_M'

     51 |     printf("ZGEMM3M_UNROLL_M=%d\n", DGEMM_DEFAULT_UNROLL_M);
        |                                     ^

  
  /home/lz/Codes/vcpkg/buildtrees/openblas/src/v0.3.27-2f0a3701cf.clean/getarch_2nd.c:57:37:
  error: use of undeclared identifier 'DGEMM_DEFAULT_UNROLL_N'

     57 |     printf("ZGEMM3M_UNROLL_N=%d\n", DGEMM_DEFAULT_UNROLL_N);
        |                                     ^

  
  /home/lz/Codes/vcpkg/buildtrees/openblas/src/v0.3.27-2f0a3701cf.clean/getarch_2nd.c:79:50:
  error: use of undeclared identifier 'SGEMM_DEFAULT_Q'

     79 |     printf("#define SLOCAL_BUFFER_SIZE\t%ld\n", (SGEMM_DEFAULT_Q * SGEMM_DEFAULT_UNROLL_N * 4 * 1 *  sizeof(float)));
        |                                                  ^

  
  /home/lz/Codes/vcpkg/buildtrees/openblas/src/v0.3.27-2f0a3701cf.clean/getarch_2nd.c:79:68:
  error: use of undeclared identifier 'SGEMM_DEFAULT_UNROLL_N'

     79 |     printf("#define SLOCAL_BUFFER_SIZE\t%ld\n", (SGEMM_DEFAULT_Q * SGEMM_DEFAULT_UNROLL_N * 4 * 1 *  sizeof(float)));
        |                                                                    ^

  
  /home/lz/Codes/vcpkg/buildtrees/openblas/src/v0.3.27-2f0a3701cf.clean/getarch_2nd.c:80:50:
  error: use of undeclared identifier 'DGEMM_DEFAULT_Q'

     80 |     printf("#define DLOCAL_BUFFER_SIZE\t%ld\n", (DGEMM_DEFAULT_Q * DGEMM_DEFAULT_UNROLL_N * 2 * 1 *  sizeof(double)));
        |                                                  ^

  
  /home/lz/Codes/vcpkg/buildtrees/openblas/src/v0.3.27-2f0a3701cf.clean/getarch_2nd.c:80:68:
  error: use of undeclared identifier 'DGEMM_DEFAULT_UNROLL_N'

     80 |     printf("#define DLOCAL_BUFFER_SIZE\t%ld\n", (DGEMM_DEFAULT_Q * DGEMM_DEFAULT_UNROLL_N * 2 * 1 *  sizeof(double)));
        |                                                                    ^

  
  /home/lz/Codes/vcpkg/buildtrees/openblas/src/v0.3.27-2f0a3701cf.clean/getarch_2nd.c:81:50:
  error: use of undeclared identifier 'CGEMM_DEFAULT_Q'

     81 |     printf("#define CLOCAL_BUFFER_SIZE\t%ld\n", (CGEMM_DEFAULT_Q * CGEMM_DEFAULT_UNROLL_N * 4 * 2 *  sizeof(float)));
        |                                                  ^

  
  /home/lz/Codes/vcpkg/buildtrees/openblas/src/v0.3.27-2f0a3701cf.clean/getarch_2nd.c:81:68:
  error: use of undeclared identifier 'CGEMM_DEFAULT_UNROLL_N'

     81 |     printf("#define CLOCAL_BUFFER_SIZE\t%ld\n", (CGEMM_DEFAULT_Q * CGEMM_DEFAULT_UNROLL_N * 4 * 2 *  sizeof(float)));
        |                                                                    ^

  
  /home/lz/Codes/vcpkg/buildtrees/openblas/src/v0.3.27-2f0a3701cf.clean/getarch_2nd.c:82:50:
  error: use of undeclared identifier 'ZGEMM_DEFAULT_Q'

     82 |     printf("#define ZLOCAL_BUFFER_SIZE\t%ld\n", (ZGEMM_DEFAULT_Q * ZGEMM_DEFAULT_UNROLL_N * 2 * 2 *  sizeof(double)));
        |                                                  ^

  fatal error: too many errors emitted, stopping now [-ferror-limit=]

  20 errors generated.

  ninja: build stopped: subcommand failed.



Call Stack (most recent call first):
  cmake/system.cmake:170 (include)
  CMakeLists.txt:104 (include)


-- Configuring incomplete, errors occurred!
[2/2] "/home/lz/cmake-3.30.3-linux-x86_64/bin/cmake" -E chdir "../../x64-linux-dbg" "/home/lz/cmake-3.30.3-linux-x86_64/bin/cmake" "/home/lz/Codes/vcpkg/buildtrees/openblas/src/v0.3.27-2f0a3701cf.clean" "-G" "Ninja" "-DCMAKE_BUILD_TYPE=Debug" "-DCMAKE_INSTALL_PREFIX=/home/lz/Codes/vcpkg/packages/openblas_x64-linux/debug" "-DFETCHCONTENT_FULLY_DISCONNECTED=ON" "-DUSE_THREAD=OFF" "-DUSE_SIMPLE_THREADED_LEVEL3=OFF" "-DDYNAMIC_ARCH=OFF" "-DBUILD_WITHOUT_LAPACK=ON" "-DNOFORTRAN=ON" "-DBU=_" "-DCMAKE_MAKE_PROGRAM=/home/lz/Codes/vcpkg/downloads/tools/ninja/1.10.2-linux/ninja" "-DCMAKE_SYSTEM_NAME=Linux" "-DBUILD_SHARED_LIBS=OFF" "-DVCPKG_CHAINLOAD_TOOLCHAIN_FILE=/home/lz/Codes/vcpkg/scripts/toolchains/linux.cmake" "-DVCPKG_TARGET_TRIPLET=x64-linux" "-DVCPKG_SET_CHARSET_FLAG=ON" "-DVCPKG_PLATFORM_TOOLSET=external" "-DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_SYSTEM_PACKAGE_REGISTRY=ON" "-DCMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP=TRUE" "-DCMAKE_VERBOSE_MAKEFILE=ON" "-DVCPKG_APPLOCAL_DEPS=OFF" "-DCMAKE_TOOLCHAIN_FILE=/home/lz/Codes/vcpkg/scripts/buildsystems/vcpkg.cmake" "-DCMAKE_ERROR_ON_ABSOLUTE_INSTALL_DESTINATION=ON" "-DVCPKG_CXX_FLAGS= -DNEEDBUNDERSCORE" "-DVCPKG_CXX_FLAGS_RELEASE=" "-DVCPKG_CXX_FLAGS_DEBUG=" "-DVCPKG_C_FLAGS= -DNEEDBUNDERSCORE" "-DVCPKG_C_FLAGS_RELEASE=" "-DVCPKG_C_FLAGS_DEBUG=" "-DVCPKG_CRT_LINKAGE=dynamic" "-DVCPKG_LINKER_FLAGS=" "-DVCPKG_LINKER_FLAGS_RELEASE=" "-DVCPKG_LINKER_FLAGS_DEBUG=" "-DVCPKG_TARGET_ARCHITECTURE=x64" "-DCMAKE_INSTALL_LIBDIR:STRING=lib" "-DCMAKE_INSTALL_BINDIR:STRING=bin" "-D_VCPKG_ROOT_DIR=/home/lz/Codes/vcpkg" "-D_VCPKG_INSTALLED_DIR=/home/lz/Codes/colmap/build/vcpkg_installed" "-DVCPKG_MANIFEST_INSTALL=OFF"
FAILED: ../../x64-linux-dbg/CMakeCache.txt 
"/home/lz/cmake-3.30.3-linux-x86_64/bin/cmake" -E chdir "../../x64-linux-dbg" "/home/lz/cmake-3.30.3-linux-x86_64/bin/cmake" "/home/lz/Codes/vcpkg/buildtrees/openblas/src/v0.3.27-2f0a3701cf.clean" "-G" "Ninja" "-DCMAKE_BUILD_TYPE=Debug" "-DCMAKE_INSTALL_PREFIX=/home/lz/Codes/vcpkg/packages/openblas_x64-linux/debug" "-DFETCHCONTENT_FULLY_DISCONNECTED=ON" "-DUSE_THREAD=OFF" "-DUSE_SIMPLE_THREADED_LEVEL3=OFF" "-DDYNAMIC_ARCH=OFF" "-DBUILD_WITHOUT_LAPACK=ON" "-DNOFORTRAN=ON" "-DBU=_" "-DCMAKE_MAKE_PROGRAM=/home/lz/Codes/vcpkg/downloads/tools/ninja/1.10.2-linux/ninja" "-DCMAKE_SYSTEM_NAME=Linux" "-DBUILD_SHARED_LIBS=OFF" "-DVCPKG_CHAINLOAD_TOOLCHAIN_FILE=/home/lz/Codes/vcpkg/scripts/toolchains/linux.cmake" "-DVCPKG_TARGET_TRIPLET=x64-linux" "-DVCPKG_SET_CHARSET_FLAG=ON" "-DVCPKG_PLATFORM_TOOLSET=external" "-DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_SYSTEM_PACKAGE_REGISTRY=ON" "-DCMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP=TRUE" "-DCMAKE_VERBOSE_MAKEFILE=ON" "-DVCPKG_APPLOCAL_DEPS=OFF" "-DCMAKE_TOOLCHAIN_FILE=/home/lz/Codes/vcpkg/scripts/buildsystems/vcpkg.cmake" "-DCMAKE_ERROR_ON_ABSOLUTE_INSTALL_DESTINATION=ON" "-DVCPKG_CXX_FLAGS= -DNEEDBUNDERSCORE" "-DVCPKG_CXX_FLAGS_RELEASE=" "-DVCPKG_CXX_FLAGS_DEBUG=" "-DVCPKG_C_FLAGS= -DNEEDBUNDERSCORE" "-DVCPKG_C_FLAGS_RELEASE=" "-DVCPKG_C_FLAGS_DEBUG=" "-DVCPKG_CRT_LINKAGE=dynamic" "-DVCPKG_LINKER_FLAGS=" "-DVCPKG_LINKER_FLAGS_RELEASE=" "-DVCPKG_LINKER_FLAGS_DEBUG=" "-DVCPKG_TARGET_ARCHITECTURE=x64" "-DCMAKE_INSTALL_LIBDIR:STRING=lib" "-DCMAKE_INSTALL_BINDIR:STRING=bin" "-D_VCPKG_ROOT_DIR=/home/lz/Codes/vcpkg" "-D_VCPKG_INSTALLED_DIR=/home/lz/Codes/colmap/build/vcpkg_installed" "-DVCPKG_MANIFEST_INSTALL=OFF"
CMake Deprecation Warning at CMakeLists.txt:5 (cmake_minimum_required):
  Compatibility with CMake < 3.5 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.


-- The C compiler identification is Clang 18.1.8
-- The ASM compiler identification is Clang with GNU-like command-line
-- Found assembler: /usr/bin/clang
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/clang - skipped
-- Detecting C compile features
-- Detecting C compile features - done
CMake Warning at CMakeLists.txt:101 (message):
  CMake support is experimental.  It does not yet support all build options
  and may not produce the same Makefiles that OpenBLAS ships with.


CMake Warning (dev) at cmake/system_check.cmake:13 (if):
  Policy CMP0054 is not set: Only interpret if() arguments as variables or
  keywords when unquoted.  Run "cmake --help-policy CMP0054" for policy
  details.  Use the cmake_policy command to set the policy and suppress this
  warning.

  Quoted variables like "LINUX" will no longer be dereferenced when the
  policy is set to NEW.  Since the policy is not set the OLD behavior will be
  used.
Call Stack (most recent call first):
  cmake/system.cmake:8 (include)
  CMakeLists.txt:104 (include)
This warning is for project developers.  Use -Wno-dev to suppress it.

-- Looking for stdatomic.h
-- Looking for stdatomic.h - found
-- GEMM multithread threshold set to 4.
-- Running getarch
-- GETARCH results:
CORE=HASWELL
LIBCORE=haswell
NUM_CORES=22
HAVE_MMX=1
HAVE_SSE=1
HAVE_SSE2=1
HAVE_SSE3=1
HAVE_SSSE3=1
HAVE_SSE4_1=1
HAVE_SSE4_2=1
HAVE_AVX=1
HAVE_AVX2=1
HAVE_FMA3=1
MAKEFLAGS += -j 22

CMake Error at cmake/prebuild.cmake:1429 (MESSAGE):
  Compiling getarch_2nd failed Change Dir:
  '/home/lz/Codes/vcpkg/buildtrees/openblas/x64-linux-dbg/getarch2_build/CMakeFiles/CMakeTmp'




  Run Build Command(s):
  /home/lz/Codes/vcpkg/downloads/tools/ninja/1.10.2-linux/ninja -v cmTC_89ea6

  [1/2] /usr/bin/clang -fPIC -DNEEDBUNDERSCORE -march=native
  -DGEMM_MULTITHREAD_THRESHOLD=4 -g -DNO_PARALLEL_MAKE=0
  -I"/home/lz/Codes/vcpkg/buildtrees/openblas/x64-linux-dbg/getarch2_build"
  -I"/home/lz/Codes/vcpkg/buildtrees/openblas/src/v0.3.27-2f0a3701cf.clean"
  -I"/home/lz/Codes/vcpkg/buildtrees/openblas/x64-linux-dbg" -MD -MT
  CMakeFiles/cmTC_89ea6.dir/getarch_2nd.c.o -MF
  CMakeFiles/cmTC_89ea6.dir/getarch_2nd.c.o.d -o
  CMakeFiles/cmTC_89ea6.dir/getarch_2nd.c.o -c
  /home/lz/Codes/vcpkg/buildtrees/openblas/src/v0.3.27-2f0a3701cf.clean/getarch_2nd.c


  FAILED: CMakeFiles/cmTC_89ea6.dir/getarch_2nd.c.o

  /usr/bin/clang -fPIC -DNEEDBUNDERSCORE -march=native
  -DGEMM_MULTITHREAD_THRESHOLD=4 -g -DNO_PARALLEL_MAKE=0
  -I"/home/lz/Codes/vcpkg/buildtrees/openblas/x64-linux-dbg/getarch2_build"
  -I"/home/lz/Codes/vcpkg/buildtrees/openblas/src/v0.3.27-2f0a3701cf.clean"
  -I"/home/lz/Codes/vcpkg/buildtrees/openblas/x64-linux-dbg" -MD -MT
  CMakeFiles/cmTC_89ea6.dir/getarch_2nd.c.o -MF
  CMakeFiles/cmTC_89ea6.dir/getarch_2nd.c.o.d -o
  CMakeFiles/cmTC_89ea6.dir/getarch_2nd.c.o -c
  /home/lz/Codes/vcpkg/buildtrees/openblas/src/v0.3.27-2f0a3701cf.clean/getarch_2nd.c



  /home/lz/Codes/vcpkg/buildtrees/openblas/src/v0.3.27-2f0a3701cf.clean/getarch_2nd.c:22:35:
  error: use of undeclared identifier 'SGEMM_DEFAULT_UNROLL_M'

     22 |     printf("SGEMM_UNROLL_M=%d\n", SGEMM_DEFAULT_UNROLL_M);
        |                                   ^

  
  /home/lz/Codes/vcpkg/buildtrees/openblas/src/v0.3.27-2f0a3701cf.clean/getarch_2nd.c:23:35:
  error: use of undeclared identifier 'SGEMM_DEFAULT_UNROLL_N'

     23 |     printf("SGEMM_UNROLL_N=%d\n", SGEMM_DEFAULT_UNROLL_N);
        |                                   ^

  
  /home/lz/Codes/vcpkg/buildtrees/openblas/src/v0.3.27-2f0a3701cf.clean/getarch_2nd.c:24:35:
  error: use of undeclared identifier 'DGEMM_DEFAULT_UNROLL_M'

     24 |     printf("DGEMM_UNROLL_M=%d\n", DGEMM_DEFAULT_UNROLL_M);
        |                                   ^

  
  /home/lz/Codes/vcpkg/buildtrees/openblas/src/v0.3.27-2f0a3701cf.clean/getarch_2nd.c:25:35:
  error: use of undeclared identifier 'DGEMM_DEFAULT_UNROLL_N'

     25 |     printf("DGEMM_UNROLL_N=%d\n", DGEMM_DEFAULT_UNROLL_N);
        |                                   ^

  
  /home/lz/Codes/vcpkg/buildtrees/openblas/src/v0.3.27-2f0a3701cf.clean/getarch_2nd.c:29:35:
  error: use of undeclared identifier 'CGEMM_DEFAULT_UNROLL_M'

     29 |     printf("CGEMM_UNROLL_M=%d\n", CGEMM_DEFAULT_UNROLL_M);
        |                                   ^

  
  /home/lz/Codes/vcpkg/buildtrees/openblas/src/v0.3.27-2f0a3701cf.clean/getarch_2nd.c:30:35:
  error: use of undeclared identifier 'CGEMM_DEFAULT_UNROLL_N'

     30 |     printf("CGEMM_UNROLL_N=%d\n", CGEMM_DEFAULT_UNROLL_N);
        |                                   ^

  
  /home/lz/Codes/vcpkg/buildtrees/openblas/src/v0.3.27-2f0a3701cf.clean/getarch_2nd.c:31:35:
  error: use of undeclared identifier 'ZGEMM_DEFAULT_UNROLL_M'

     31 |     printf("ZGEMM_UNROLL_M=%d\n", ZGEMM_DEFAULT_UNROLL_M);
        |                                   ^

  
  /home/lz/Codes/vcpkg/buildtrees/openblas/src/v0.3.27-2f0a3701cf.clean/getarch_2nd.c:32:35:
  error: use of undeclared identifier 'ZGEMM_DEFAULT_UNROLL_N'

     32 |     printf("ZGEMM_UNROLL_N=%d\n", ZGEMM_DEFAULT_UNROLL_N);
        |                                   ^

  
  /home/lz/Codes/vcpkg/buildtrees/openblas/src/v0.3.27-2f0a3701cf.clean/getarch_2nd.c:39:37:
  error: use of undeclared identifier 'SGEMM_DEFAULT_UNROLL_M'

     39 |     printf("CGEMM3M_UNROLL_M=%d\n", SGEMM_DEFAULT_UNROLL_M);
        |                                     ^

  
  /home/lz/Codes/vcpkg/buildtrees/openblas/src/v0.3.27-2f0a3701cf.clean/getarch_2nd.c:45:37:
  error: use of undeclared identifier 'SGEMM_DEFAULT_UNROLL_N'

     45 |     printf("CGEMM3M_UNROLL_N=%d\n", SGEMM_DEFAULT_UNROLL_N);
        |                                     ^

  
  /home/lz/Codes/vcpkg/buildtrees/openblas/src/v0.3.27-2f0a3701cf.clean/getarch_2nd.c:51:37:
  error: use of undeclared identifier 'DGEMM_DEFAULT_UNROLL_M'

     51 |     printf("ZGEMM3M_UNROLL_M=%d\n", DGEMM_DEFAULT_UNROLL_M);
        |                                     ^

  
  /home/lz/Codes/vcpkg/buildtrees/openblas/src/v0.3.27-2f0a3701cf.clean/getarch_2nd.c:57:37:
  error: use of undeclared identifier 'DGEMM_DEFAULT_UNROLL_N'

     57 |     printf("ZGEMM3M_UNROLL_N=%d\n", DGEMM_DEFAULT_UNROLL_N);
        |                                     ^

  
  /home/lz/Codes/vcpkg/buildtrees/openblas/src/v0.3.27-2f0a3701cf.clean/getarch_2nd.c:79:50:
  error: use of undeclared identifier 'SGEMM_DEFAULT_Q'

     79 |     printf("#define SLOCAL_BUFFER_SIZE\t%ld\n", (SGEMM_DEFAULT_Q * SGEMM_DEFAULT_UNROLL_N * 4 * 1 *  sizeof(float)));
        |                                                  ^

  
  /home/lz/Codes/vcpkg/buildtrees/openblas/src/v0.3.27-2f0a3701cf.clean/getarch_2nd.c:79:68:
  error: use of undeclared identifier 'SGEMM_DEFAULT_UNROLL_N'

     79 |     printf("#define SLOCAL_BUFFER_SIZE\t%ld\n", (SGEMM_DEFAULT_Q * SGEMM_DEFAULT_UNROLL_N * 4 * 1 *  sizeof(float)));
        |                                                                    ^

  
  /home/lz/Codes/vcpkg/buildtrees/openblas/src/v0.3.27-2f0a3701cf.clean/getarch_2nd.c:80:50:
  error: use of undeclared identifier 'DGEMM_DEFAULT_Q'

     80 |     printf("#define DLOCAL_BUFFER_SIZE\t%ld\n", (DGEMM_DEFAULT_Q * DGEMM_DEFAULT_UNROLL_N * 2 * 1 *  sizeof(double)));
        |                                                  ^

  
  /home/lz/Codes/vcpkg/buildtrees/openblas/src/v0.3.27-2f0a3701cf.clean/getarch_2nd.c:80:68:
  error: use of undeclared identifier 'DGEMM_DEFAULT_UNROLL_N'

     80 |     printf("#define DLOCAL_BUFFER_SIZE\t%ld\n", (DGEMM_DEFAULT_Q * DGEMM_DEFAULT_UNROLL_N * 2 * 1 *  sizeof(double)));
        |                                                                    ^

  
  /home/lz/Codes/vcpkg/buildtrees/openblas/src/v0.3.27-2f0a3701cf.clean/getarch_2nd.c:81:50:
  error: use of undeclared identifier 'CGEMM_DEFAULT_Q'

     81 |     printf("#define CLOCAL_BUFFER_SIZE\t%ld\n", (CGEMM_DEFAULT_Q * CGEMM_DEFAULT_UNROLL_N * 4 * 2 *  sizeof(float)));
        |                                                  ^

  
  /home/lz/Codes/vcpkg/buildtrees/openblas/src/v0.3.27-2f0a3701cf.clean/getarch_2nd.c:81:68:
  error: use of undeclared identifier 'CGEMM_DEFAULT_UNROLL_N'

     81 |     printf("#define CLOCAL_BUFFER_SIZE\t%ld\n", (CGEMM_DEFAULT_Q * CGEMM_DEFAULT_UNROLL_N * 4 * 2 *  sizeof(float)));
        |                                                                    ^

  
  /home/lz/Codes/vcpkg/buildtrees/openblas/src/v0.3.27-2f0a3701cf.clean/getarch_2nd.c:82:50:
  error: use of undeclared identifier 'ZGEMM_DEFAULT_Q'

     82 |     printf("#define ZLOCAL_BUFFER_SIZE\t%ld\n", (ZGEMM_DEFAULT_Q * ZGEMM_DEFAULT_UNROLL_N * 2 * 2 *  sizeof(double)));
        |                                                  ^

  fatal error: too many errors emitted, stopping now [-ferror-limit=]

  20 errors generated.

  ninja: build stopped: subcommand failed.



Call Stack (most recent call first):
  cmake/system.cmake:170 (include)
  CMakeLists.txt:104 (include)


-- Configuring incomplete, errors occurred!
ninja: build stopped: subcommand failed.
```

**Comment by martin-frbg** [comment]
@HernandoR that looks to be a bit different from the original issue, I'm not quite sure what to make of it. getarch_2nd.c includes both the "config.h" that was generated by the `getarch` program and the "param.h" from the toplevel source directory - param.h has the SGEMM_DEFAULT_UNROLL_N etc  for each cpu type and config.h tells which cpu entry to look in. From the cmake output getarch must have run successfully, so there should be a config.h in your build folder (and I think an identical copy in the getarch2nd_build folder beneath that). Can you please check that it is there and contains the line `#define HASWELL` near the top ?
Maybe there is some kind of permission problem on your filesystem (or the disk is full ?) leading to the file not getting written, the failed CMakeCache.txt access and failed ChangeDir messages look a bit suspicious too...

**Comment by HernandoR** [comment]
@martin-frbg the build for openblas is trigered attomatically by vcpkg, and can easily reperformed (at least in my env) through `vcpkg install openblas`
i can confirm that the dir and config.h is existed, and can be accessed, at least when i instect into, after the break. 

```
╰─❯ tree getarch2_build -p
[drwxr-xr-x]  getarch2_build
├── [drwxr-xr-x]  CMakeFiles
│   └── [drwxr-xr-x]  CMakeTmp
└── [-rw-r--r--]  config.h
```
the `config.h` is as follows. and i'm hosting in a x64-linux targeting for x64-linux:
```
cat /home/lz/Codes/vcpkg/buildtrees/openblas/x64-linux-rel/getarch2_build/config.h
#define OS_LINUX        1
#define ARCH_X86_64     1
#define C_GCC   1
#define __64BIT__       1
#define FUNDERSCORE
#define BUNDERSCORE _
#define NEEDBUNDERSCORE 1
#define INTEL_UNKNOWN
#define L1_CODE_SIZE 65536
#define L1_CODE_ASSOCIATIVE 16
#define L1_CODE_LINESIZE 64
#define L1_DATA_SIZE 49152
#define L1_DATA_ASSOCIATIVE 12
#define L1_DATA_LINESIZE 64
#define L2_SIZE 2097152
#define L2_ASSOCIATIVE 7
#define L2_LINESIZE 64
#define DTB_DEFAULT_ENTRIES 32
#define HAVE_CMOV
#define HAVE_MMX
#define HAVE_SSE
#define HAVE_SSE2
#define HAVE_SSE3
#define HAVE_SSSE3
#define HAVE_SSE4_1
#define HAVE_SSE4_2
#define HAVE_AVX
#define HAVE_AVX2
#define HAVE_FMA3
#define HAVE_CFLUSH
#define NUM_SHAREDCACHE 2
#define NUM_CORES 16
#define CORE_HASWELL
#define CHAR_CORENAME "HASWELL"
```
Let me know if you think i should raise a new issue.

**Comment by martin-frbg** [comment]
The `#define INTEL_UNKNOWN` is the problem - strange as getarch clearly managed to recognize it as CORE_HASWELL. Probably a wrong or missing entry in cpuid_x86.c - what is your actual cpu model please ?

**Comment by HernandoR** [comment]
CPU CPU  Intel(R) Core(TM) Ultra 7 155H, and hosted in WSL2, 
Neofetch is able to recognize the cpu model 
```
            .-/+oossssoo+/-.               lz@Mi-liuzhen19
        `:+ssssssssssssssssss+:`           ---------------
      -+ssssssssssssssssssyyssss+-         OS: Ubuntu 24.04 LTS on Wind
    .ossssssssssssssssssdMMMNysssso.       Kernel: 5.15.153.1-microsoft
   /ssssssssssshdmmNNmmyNMMMMhssssss/      Uptime: 1 hour, 22 mins
  +ssssssssshmydMMMMMMMNddddyssssssss+     Packages: 716 (dpkg), 32 (br
 /sssssssshNMMMyhhyyyyhmNMMMNhssssssss/    Shell: zsh 5.9
.ssssssssdMMMNhsssssssssshNMMMdssssssss.   Theme: Adwaita [GTK3]
+sssshhhyNMMNyssssssssssssyNMMMysssssss+   Icons: Adwaita [GTK3]
ossyNMMMNyMMhsssssssssssssshmmmhssssssso   Terminal: Windows Terminal
ossyNMMMNyMMhsssssssssssssshmmmhssssssso   CPU: Intel Ultra 7 155H (22)
+sssshhhyNMMNyssssssssssssyNMMMysssssss+   GPU: b0f3:00:00.0 Microsoft
.ssssssssdMMMNhsssssssssshNMMMdssssssss.   Memory: 1460MiB / 15813MiB
 /sssssssshNMMMyhhyyyyhdNMMMNhssssssss/
  +sssssssssdmydMMMMMMMMddddyssssssss+
   /ssssssssssshdmNNNNmyNMMMMhssssss/
    .ossssssssssssssssssdMMMNysssso.
      -+sssssssssssssssssyyyssss+-
        `:+ssssssssssssssssss+:`
            .-/+oossssoo+/-.

```

**Comment by martin-frbg** [comment]
Ah that's unfortunate - I've got the same "Meteor Lake" cpu and added its cpuid in #4783, but that fix for cpu detection was released only in 0.3.28 while you are trying to build 0.3.27. You can either upgrade to 0.3.28 or copy the file cpuid_x86.c from the "Code" page here on github.

**Comment by HernandoR** [comment]
> Ah that's unfortunate - I've got the same "Meteor Lake" cpu and added its cpuid in #4783, but that fix for cpu detection was released only in 0.3.28 while you are trying to build 0.3.27. You can either upgrade to 0.3.28 or copy the file cpuid_x86.c from the "Code" page here on github.

Cool! I'm using vcpkg which has the latest version of 0.3.27, not sure who is maintaining the port. I'll use cmake fetch contents for now.
